index.bs

```diff
@@ -8852,15 +8852,13 @@ specified sequence of user input actions.
       )
 
       input.AngleProperties = (
-        ; 0 .. Math.PI / 2
-        ? altitudeAngle: (0.0..1.5707963267948966) .default 0.0,
-        ; 0 .. 2 * Math.PI
-        ? azimuthAngle: (0.0..6.283185307179586) .default 0.0,
+         ? altitudeAngle: float .default 0.0,
+         ? azimuthAngle: float .default 0.0,
       )
 
       input.TiltProperties = (
```

Comment:
Because `AngleProperties` does not validate that `tiltX` exists.

Reply:
Would an alternative fix be putting input.AngleProperties and input.TiltProperties each into a dedicated key in PointerCommonProperties?

Reply:
Yes, that will fix this as well.

If you want to increase the scope of this proposal, there are other things to consider then:

 - `tiltX` and `tiltY` are considered worse versions of `altitudeAngle` and `azimuthAngle` (https://github.com/w3c/pointerevents/issues/274), so it may be worth just _removing_ `tiltX` and `tiltY`.
   - Note this is possible because there is a standardized algorithm here: https://w3c.github.io/pointerevents/#converting-between-tiltx-tilty-and-altitudeangle-azimuthangle

Reply:
Alternatively, we can get rid of the type union and just do
```
igin: input.Origin .default "viewport",
      }

      input.PointerCommonProperties = (
        ? width: js-uint .default 1,
        ? height: js-uint .default 1,
        ? pressure: float .default 0.0,
        ? tangentialPressure: float .default 0.0,
        ? twist: (0..359) .default 0,
        input.TiltProperties,
        input.AngleProperties
      )

      input.AngleProperties = (
        ; 0 .. Math.PI / 2
        ? altitudeAngle: (0.0..1.5707963267948966) .default 0.0,
        ; 0 .. 2 * Math.PI
        ? azimuthAngle: (0.0..6.283185307179586) .default 0.0,
      )

      input.TiltProperties = (
        ? tiltX: (-90..90) .default 0,
        ? tiltY: (-90..90) .default 0,
      )
```

Technically, this is how the classic spec is defined, not using the type union IIRC.

Reply:
@jgraham @juliandescottes what's your take on this issue?

I think removing ranges is acceptable but on the other hand it's nice to have ranges in CDDL, so perhaps we could go for one of the other two alternatives?